numpy/typing/mypy_plugin.py

```diff
@@ -186,15 +192,12 @@ def get_additional_deps(
     def plugin(version: str) -> type:
         import warnings
 
-        warnings.warn(
-            "`numpy.typing.mypy_plugin` is deprecated, and will be removed in "
-            "a future release.",
-            category=DeprecationWarning,
-            stacklevel=3,
+        plugin = "numpy.typing.mypy_plugin"
```

Comment:
```suggestion
        plugin = "numpy.typing.mypy_plugin"
        # Deprecated 2025-01-10, NumPy 2.3
```
Just to make it easy to grep.